### PR TITLE
Remove dangerous default executor from HandledTransportAction

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today subclasses of `HandledTransportAction` can specify the executor on
which they run, but the executor is optional and if omitted will use
`DIRECT_EXECUTOR_SERVICE`, which means the action runs on a transport
thread. This is a dangerous default behaviour because it makes it easy
to add new transport actions which implicitly run on a network thread,
which is very hard to pick up in reviews.

This commit makes the executor explicit in all callers, and marks the
dangerous methods for removal.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/part-2

**Comment by DaveCTurner** [comment]
NB this was almost completely mechanical, just automatically inlining the bad methods within `HandledTransportAction` and then bringing them back to mark them as deprecated. However then I realised that this added the unnecessary `canTripCircuitBreaker` parameter to all callers too so I did another almost-completely mechanical change to remove that in 0f18e27.

**Review by juliaElastic — APPROVED** [review]
Fleet change LGTM

**Review by DiannaHohensee — APPROVED** [review]
Nice, much more obvious :)

File: server/src/main/java/org/elasticsearch/action/support/HandledTransportAction.java [hunk 1]
```diff
@@ -11,20 +11,21 @@
...
+ * A {@link TransportAction} which registers a handler for itself with the transport service.
  */
 public abstract class HandledTransportAction<Request extends ActionRequest, Response extends ActionResponse> extends TransportAction<
     Request,
     Response> {
 
+    /**
+     * @deprecated always specify an executor
+     */
+    @Deprecated(forRemoval = true)
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/support/HandledTransportAction.java:28]: Right

File: server/src/main/java/org/elasticsearch/action/support/HandledTransportAction.java [hunk 2]
```diff
@@ -11,20 +11,21 @@
...
-import org.elasticsearch.tasks.Task;
-import org.elasticsearch.transport.TransportChannel;
-import org.elasticsearch.transport.TransportRequestHandler;
 import org.elasticsearch.transport.TransportService;
 
 import java.util.concurrent.Executor;
 
 /**
- * A TransportAction that self registers a handler into the transport service
+ * A {@link TransportAction} which registers a handler for itself with the transport service.
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/support/HandledTransportAction.java]: Hmm yeah in a sense what other action could we really be registering here? But we do register subsidiary handlers for actions with names like `actionName + "[x]"` elsewhere. I changed it to make it clearer that we're not mangling the name at all with the registration.

[the remark this answers is not in the document]
Comment (DiannaHohensee) [reply server/src/main/java/org/elasticsearch/action/support/HandledTransportAction.java]: Gotcha, that's a better lead on what it's doing, thank you.

I don't think I myself have an understanding of what registering action names means in the bigger picture, but that's not something to be documented in this patch.

[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/support/HandledTransportAction.java]: It's not very well documented today; I opened https://github.com/elastic/elasticsearch/issues/100229

[the remark this answers is not in the document]
Comment (DiannaHohensee) [reply server/src/main/java/org/elasticsearch/action/support/HandledTransportAction.java]: Great, thank you!

**Pull request merged**